fhir-bulkimportexport-webapp/src/main/java/com/ibm/fhir/bulkcommon/BulkDataUtils.java

```diff
@@ -145,101 +157,159 @@ private static int getFhirResourceFromBufferReader(BufferedReader resReader, int
-                exported++;
-                if (exported == Constants.IMPORT_NUMOFFHIRRESOURCES_PERREAD) {
-                    break;
+                try {
+                    fhirResources.add(FHIRParser.parser(Format.JSON).parse(new StringReader(resLine)));
+                    exported++;
+                    if (exported == Constants.IMPORT_NUMOFFHIRRESOURCES_PERREAD) {
+                        break;
+                    }
+                } catch (FHIRParserException e) {
+                    // Log and skip the invalid FHIR resource.
+                    logger.warning("getFhirResourceFromBufferReader: " + e.getMessage());
+                    logger.warning("getFhirResourceFromBufferReader: " + "Failed to parse '" + resLine + "'");
+                    parseFailures++;
+                    continue;
                 }
             }
         } while (resLine != null);
-        return exported;
+        return parseFailures;
     }
 
-    public static int readFhirResourceFromObjectStore(AmazonS3 cosClient, String bucketName, String itemName,
-           int numOfLinesToSkip, List<Resource> fhirResources, boolean isReuseInput, ImportTransientUserData transientUserData) {
-        int exported = 0;
-        if (isReuseInput) {
-            if (transientUserData.getBufferReader() == null) {
-                S3Object item = cosClient.getObject(new GetObjectRequest(bucketName, itemName));
-                S3ObjectInputStream s3InStream = item.getObjectContent();
-                transientUserData.setInputStream(s3InStream);
-                BufferedReader resReader = new BufferedReader(new InputStreamReader(s3InStream));
-                transientUserData.setBufferReader(resReader);
-            }
-            try {
-                exported = getFhirResourceFromBufferReader(transientUserData.getBufferReader(), 0, fhirResources);
-            } catch (Exception ioe) {
-                logger.warning("readFhirResourceFromObjectStore: " + "Error proccesing file " + itemName + " - " + ioe.getMessage());
-                exported = 0;
-            }
-
-        } else {
-            S3Object item = cosClient.getObject(new GetObjectRequest(bucketName, itemName));
-            try (S3ObjectInputStream s3InStream = item.getObjectContent();
-                 BufferedReader resReader = new BufferedReader(new InputStreamReader(s3InStream))) {
-                exported = getFhirResourceFromBufferReader(resReader, numOfLinesToSkip, fhirResources);
-                // Notify s3 client to abort and prevent the server from keeping on sending data.
-                s3InStream.abort();
-            } catch (Exception ioe) {
-                logger.warning("readFhirResourceFromObjectStore: " + "Error proccesing file " + itemName + " - " + ioe.getMessage());
-                exported = 0;
+    public static void cleanup4TransientUserData(ImportTransientUserData transientUserData, boolean isAbort) throws Exception {
```

Comment:
```suggestion
    public static void cleanupForTransientUserData(ImportTransientUserData transientUserData, boolean isAbort) throws Exception {
```

I don't think we should use the number 4 here

Reply:
haha, good point! will change

Reply:
done